coreclient/src/clients/add_contact.rs

```diff
@@ -129,6 +128,20 @@ impl CoreUser {
     ) -> anyhow::Result<ChatId> {
         let client = self.api_client()?;
 
+        // Phase 0: Sanity checks
+        // Check whether we already have this user as a contact
+        if self.contact(&user_id).await.is_some() {
+            bail!("User is already a contact");
```

Comment:
Wouldn't we want to return a proper error here so we can bubble it up to the UI?